src/search.cpp

```diff
@@ -143,6 +145,11 @@ Search::Worker::Worker(SharedState&                    sharedState,
 }
 
 void Search::Worker::start_searching() {
+
+    // Initialize accumulator refresh entries
+    for (int i = 0; i < SQUARE_NB; i++)
+        networks.big.init_refresh_entry(rootPos.refreshTable[i]);
```

Comment:
I see that this is required because if the network changes we need to notify the cache somehow. However I don't like that the worker is responsible for this. This is also a consequence of the cache being owned by Position. I also don't like `init` functions in modern C++.

I'd suggest the following resolution:

1. Move the cache to `Worker`. It therefore sits decoupled from the position.
2. Initialize the table properly in the constructor and add a clear() function that does that same.
3. This code can stay here now as it would make a bit more sense, as the whole cache is confined to the worker.
4. Pass a reference to this cache down the callstack to the evaluation. I know, it's another parameter, but it makes sense. If there's too many parameters we'll think about some abstraction. What matters is that `Position` is not a good abstraction for this.